lib/util/profile.js

```diff
@@ -73,7 +73,7 @@ function optionToItinerary (option) {
         duration: walkOnTime,
         transitLeg: false,
         from: {
-          name: 'Origin'
+          name: locationString(query && query.from.name, 'Destination')
```

Comment:
@demory I would suggest that we put `query.from.name` and `query.to.name` into variables -- for clarity in the code and to make the whole thing less verbose.  These values are used a number of times in this function.